contract-tests/tests/test/amazon/mysqlclient/mysqlclient_test.py

```diff
@@ -0,0 +1,99 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+from typing import Dict, List
+
+from testcontainers.mysql import MySqlContainer
+from typing_extensions import override
+
+from amazon.base.contract_test_base import NETWORK_NAME
+from amazon.base.database_contract_test_base import (
+    DATABASE_HOST,
+    DATABASE_NAME,
+    DATABASE_PASSWORD,
+    DATABASE_USER,
+    LOCAL_ROOT,
+    DatabaseContractTestBase,
+)
+from amazon.utils.application_signals_constants import (
+    AWS_LOCAL_OPERATION,
+    AWS_LOCAL_SERVICE,
+    AWS_REMOTE_OPERATION,
+    AWS_REMOTE_RESOURCE_IDENTIFIER,
+    AWS_REMOTE_RESOURCE_TYPE,
+    AWS_REMOTE_SERVICE,
+    AWS_SPAN_KIND,
+)
+from opentelemetry.proto.common.v1.common_pb2 import AnyValue, KeyValue
+
+
+class MysqlClientTest(DatabaseContractTestBase):
+    @override
+    @classmethod
+    def set_up_dependency_container(cls) -> None:
+        cls.container = (
+            MySqlContainer(MYSQL_USER=DATABASE_USER, MYSQL_PASSWORD=DATABASE_PASSWORD, MYSQL_DATABASE=DATABASE_NAME)
+            .with_kwargs(network=NETWORK_NAME)
+            .with_name(DATABASE_HOST)
+        )
+        cls.container.start()
+
+    @override
+    @classmethod
+    def tear_down_dependency_container(cls) -> None:
+        cls.container.stop()
+
+    @override
+    @staticmethod
+    def get_remote_service() -> str:
+        return "mysql"
+
+    @override
+    @staticmethod
+    def get_database_port() -> int:
+        return 3306
+
+    @override
+    @staticmethod
+    def get_application_image_name() -> str:
+        return "aws-application-signals-tests-mysqlclient-app"
+
+    def test_drop_table_succeeds(self) -> None:
+        self.assert_drop_table_succeeds()
+
+    def test_create_database_succeeds(self) -> None:
+        self.assert_create_database_succeeds()
+
+    def test_fault(self) -> None:
+        self.assert_fault()
+
+    # This adapter is not currently fully supported by OTEL
+    # GitHub issue: https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1319
+    # Once the adapter is supported, we could remove _assert_aws_attributes and _assert_semantic_conventions_attributes
```

Comment:
Nitpick: We won't remove these tests, we are going to update the test cases to assert ClientSpan instead of LocalSpan and also the desired operation instead of `InternalOperation` right?

Reply:
These assertions are overriding the assertions in the [DatabaseContractTestBase class](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/main/contract-tests/tests/test/amazon/base/database_contract_test_base.py). The rest of adapters are just using the default implementation provided in DatabaseContractTestBase, so whenever this is solved in the upstream repository, we could remove the overrides, which will take the default implementation for the asserts. The final code, after the fix, should look [like this](https://github.com/aws-observability/aws-otel-python-instrumentation/blob/main/contract-tests/tests/test/amazon/base/database_contract_test_base.py).